src/scales/scale.time.js

```diff
@@ -403,6 +403,33 @@ function ticksFromTimestamps(values, majorUnit) {
 	return ticks;
 }
 
+function determineLabelFormat(data, timeOpts) {
+	var i, momentDate, hasMillis, hasTime;
+
+	if (timeOpts.tooltipFormat) {
+		return timeOpts.tooltipFormat;
+	}
+
+	// find the label with the most parts (milliseconds, minutes, etc.)
+	// format all labels with the same level of detail as the most specific label
+	for (i = 0; i < data.length; i++) {
+		momentDate = momentify(data[i], timeOpts);
+		if (momentDate.millisecond() !== 0) {
+			hasMillis = true;
+			break;
+		}
+		if (momentDate.second() !== 0 || momentDate.minute() !== 0 || momentDate.hour() !== 0) {
+			hasTime = true;
+		}
+	}
+	if (hasMillis) {
+		return 'MMM D, YYYY h:mm:ss.SSS a';
+	} else if (hasTime) {
```

Comment:
Weird that this passes the lint, `else` should not be used after a `if ... return`. Anyway, doesn't matter when removing `hasMillis` per my previous comment.

Reply:
That's because `allowElseIf: true` is the default: https://eslint.org/docs/rules/no-else-return